lib/sshutils/ctx.go

```diff
@@ -33,6 +33,13 @@ import (
 	"github.com/gravitational/teleport/lib/teleagent"
 )
 
+// TCPIPForwardDialer represents a dialer used to handle TCPIP forward requests.
+type TCPIPForwardDialer interface {
+	// Dial creates a connection to the specified tcp address. The dial operation itself
+	// is typically proxied via a child process.
+	Dial(string) (net.Conn, error)
+}
+
```

Comment:
Is this deliberately different from the "standard" dialing interface that takes a network string and an address string?